airbyte-webapp/src/components/connectorBuilder/Builder/StreamConfigView.tsx

```diff
@@ -69,53 +89,101 @@ export const StreamConfigView: React.FC<StreamConfigViewProps> = ({ streamNum })
-          tooltip="Body to attach to API requests as url-encoded form values"
-        />
-      </BuilderCard>
+      {selectedTab === "configuration" ? (
+        <>
+          <BuilderCard>
+            <BuilderField
+              type="string"
+              path={streamFieldPath("urlPath")}
+              label="Path URL"
+              tooltip="Path of the endpoint that this stream represents."
+            />
+            <BuilderField
+              type="enum"
+              path={streamFieldPath("httpMethod")}
+              options={["GET", "POST"]}
+              label="HTTP Method"
+              tooltip="HTTP method to use for requests sent to the API"
+            />
+            <BuilderField
+              type="array"
+              path={streamFieldPath("fieldPointer")}
+              label="Record selector"
+              tooltip="Pointer into the response that should be extracted as the final record"
+            />
+            <BuilderField
+              type="array"
+              path={streamFieldPath("primaryKey")}
+              label="Primary key"
+              tooltip="Pointer into the response that should be used as the primary key when deduplicating records in the destination"
+              optional
+            />
+          </BuilderCard>
+          <PaginationSection streamFieldPath={streamFieldPath} />
+          <StreamSlicerSection streamFieldPath={streamFieldPath} />
+          <BuilderCard>
+            <KeyValueListField
+              path={streamFieldPath("requestOptions.requestParameters")}
+              label="Request Parameters"
+              tooltip="Parameters to attach to API requests"
+            />
+            <KeyValueListField
+              path={streamFieldPath("requestOptions.requestHeaders")}
+              label="Request Headers"
+              tooltip="Headers to attach to API requests"
+            />
+            <KeyValueListField
+              path={streamFieldPath("requestOptions.requestBody")}
+              label="Request Body"
+              tooltip="Body to attach to API requests as url-encoded form values"
+            />
+          </BuilderCard>
+        </>
+      ) : (
+        <BuilderCard>
+          <SchemaEditor streamFieldPath={streamFieldPath} />
+        </BuilderCard>
+      )}
     </BuilderConfigView>
   );
 };
+
+const StreamTab = ({
+  selected,
+  label,
+  onSelect,
+  showErrorIndicator,
+}: {
+  selected: boolean;
+  label: string;
+  onSelect: () => void;
+  showErrorIndicator?: boolean;
+}) => (
+  <button type="button" className={classNames(styles.tab, { [styles.selectedTab]: selected })} onClick={onSelect}>
+    {label}
+    {showErrorIndicator && <Indicator />}
+  </button>
+);
+
+const SchemaEditor = ({ streamFieldPath }: { streamFieldPath: (fieldPath: string) => string }) => {
+  const [field, meta, helpers] = useField<string | undefined>(streamFieldPath("schema"));
+
+  return (
+    <>
+      <CodeEditor
+        height="500px"
```

Comment:
I don't love that this always has a static height of 500px, so I tried making some changes to make it take up the full height instead.

I got it working and put up the changes in a separate PR that is branched off of this one: https://github.com/airbytehq/airbyte/pull/21000/files
feel free to merge that into here if you agree with the approach

Reply:
Really nice improvement!